packages/expo-cli/src/commands/fetch/android.js

```diff
@@ -1,53 +1,41 @@
 import path from 'path';
-import fs from 'fs';
+import fs from 'fs-extra';
+import get from 'lodash/get';
 
-import { Credentials, Exp } from '@expo/xdl';
+import { AndroidCredentials } from '@expo/xdl';
+import { DownloadKeystore } from '../../credentials/views/AndroidCredentials';
+import { Context } from '../../credentials';
 
 import log from '../../log';
 
 async function fetchAndroidKeystoreAsync(projectDir) {
-  const {
-    args: { username, remotePackageName, remoteFullPackageName: experienceName },
-  } = await Exp.getPublishInfoAsync(projectDir);
+  const ctx = new Context();
+  await ctx.init(projectDir);
 
-  const backupKeystoreOutputPath = path.resolve(projectDir, `${remotePackageName}.jks`);
-  await Credentials.Android.backupExistingCredentials(
-    {
-      outputPath: backupKeystoreOutputPath,
-      username,
-      experienceName,
-    },
-    log
-  );
+  const backupKeystoreOutputPath = path.resolve(projectDir, `${ctx.manifest.slug}.jks`);
+
+  const view = new DownloadKeystore(ctx.manifest.slug);
```

Comment:
Maybe it would be better to pass the whole context to the constructor?

Reply:
- First argument of DownloadKeystore won't be always `ctx.manifest.slug` and sth similar can be said about most of the views.
- Context is something that should be passed along the flow of execution not kept inside some object
- Take into account that this view needs to work also as part of credential manager not only as separate instance